### ISSUE TITLE
fluid  softmax+cross_entropy是否和softmax_with_cross_entropy 等价

### ISSUE THREAD

**Issue by lgone2000** [PaddlePaddle/Paddle#11535]
下面程序 如果flag = 1 使用 就报错，flag=0 就可以跑。
对loss加mean则都能跑。不加mean时，loss输出是一个数组，这时候 opt是怎么优化的呢？
```
def testtrain1():
    flag = 1
    place = fluid.CPUPlace()
    exe = fluid.Executor(place)
    main_program = fluid.default_main_program()
    x = fluid.layers.data(name='x', shape=[256], dtype='float32')
    y = fluid.layers.data(name='y', shape=[1], dtype='int64')
    y_predict = fluid.layers.fc(input=x, size=10, act=None)
    if flag:
        prob = paddle.fluid.layers.softmax(y_predict)
        loss = paddle.fluid.layers.cross_entropy(prob, y)
    else:
        loss = fluid.layers.softmax_with_cross_entropy(logits=y_predict, label=y)
    #loss = fluid.layers.mean(loss)
    opt = fluid.optimizer.AdamOptimizer(
        learning_rate=0.001, beta1=0.9, beta2=0.999)
    opt.minimize(loss)
```

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
